### PR TITLE
fix(schedule): multi-cloud oss config error when download log bases on obcloud

### PR DESCRIPTION
**Pull request by ysjemmm**
#### What type of PR is this?
type-bug

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by yizhouxw — DISMISSED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java [hunk 1]
```diff
@@ -300,4 +308,9 @@ private InputStream forwardToDownloadLog(@NonNull String host, @NonNull Integer
             throw e;
         }
     }
+
+    private void initCloudObjectStorageService() {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java:312]: may refer JobCredentialProvider intefrace, get related CloudObjectStorage Credential

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java [hunk 2]
```diff
@@ -300,4 +308,9 @@ private InputStream forwardToDownloadLog(@NonNull String host, @NonNull Integer
...
+        Optional<ObjectStorageConfiguration> storageConfig = JobUtils.getObjectStorageConfiguration();
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java:313]: may refer JobCredentialProvider intefrace, get related CloudObjectStorage Credential

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java (hunk 1 above)
Comment (ysjemmm) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java:312]: > may refer JobCredentialProvider intefrace, get related CloudObjectStorage Credential

I will try

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java [hunk 3]
```diff
@@ -300,4 +304,11 @@ private InputStream forwardToDownloadLog(@NonNull String host, @NonNull Integer
             throw e;
         }
     }
+
+    private CloudObjectStorageService getCloudObjectStorageService(JobEntity jobEntity) {
+        JobContext jobContext = new DefaultJobContextBuilder().build(jobEntity);
+        ObjectStorageConfiguration cos = jobCredentialProvider.getCloudObjectStorageCredential(
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java:310]: cos --> objectStorageConfiguration

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java [hunk 4]
```diff
@@ -117,31 +125,26 @@ public InputStreamResource downloadLog(Long scheduleTaskId, OdcTaskLogLevel leve
...
-                        JobAttributeKeyConstants.LOG_STORAGE_BUCKET_NAME);
-                if (objId.isPresent() && bucketName.isPresent()) {
-                    return cloudObjectStorageService.generateDownloadUrl(objId.get()).toString();
-                }
-            } else {
-                log.warn("cloud object storage service is not support.");
+            String attributeKey =
+                    OdcTaskLogLevel.ALL.equals(level) ? JobAttributeKeyConstants.LOG_STORAGE_ALL_OBJECT_ID
+                            : JobAttributeKeyConstants.LOG_STORAGE_WARN_OBJECT_ID;
+            Optional<String> objId = taskFrameworkService.findByJobIdAndAttributeKey(jobId, attributeKey);
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java:143]: unessary multiple query for get attributes, may refer JobAttributeRepository.findByJobId().


we may add a `Map<String,String> getJobAttributes(Long jobId)` method in TaskFrameworkService

Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java:143]: unessary multiple query for get attributes, may refer JobAttributeRepository.findByJobId().

we may add a `Map<String,String> getJobAttributes(Long jobId)` method in TaskFrameworkService

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java (hunk 3 above)
Comment (ysjemmm) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java:310]: > cos --> objectStorageConfiguration
ok

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java (hunk 4 above)
Comment (ysjemmm) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java:143]: I know

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by guowl3 — APPROVED** [review]
lgtm

**Pull request merged**